src/hotspot/cpu/arm/c1_LIRGenerator_arm.cpp

```diff
@@ -285,7 +285,7 @@ LIR_Address* LIRGenerator::emit_array_address(LIR_Opr array_opr, LIR_Opr index_o
 }
 
 
-LIR_Opr LIRGenerator::load_immediate(int x, BasicType type) {
+LIR_Opr LIRGenerator::load_immediate(jlong x, BasicType type) {
   assert(type == T_LONG || type == T_INT, "should be");
   LIR_Opr r = make_constant(type, x);
   bool imm_in_range = AsmOperand::is_rotated_imm(x);
```

Comment:
is_rotated_imm accepts unsigned int and may return false positive

Reply:
Right. The best I can come up with is to check the `x` fits the unsigned int. See new commit. I know it builds (cross-compiles), but not sure if it passes tests.

Reply:
I still observes test failure. Have you tested on x86_32?
`----------System.err:(15/1051)----------
java.lang.IllegalStateException: Error: expected: 1040 (410), actual: 192 (c0)
        at GetObjectSizeIntrinsicsTest.assertEquals(GetObjectSizeIntrinsicsTest.java:294)
        at GetObjectSizeIntrinsicsTest.testSize_newSmallByteArray(GetObjectSizeIntrinsicsTest.java:357)`

Reply:
Yes, I did test on x86_32, see the "Additional testing" in the PR body. I just re-ran the tests on x86_32, they pass. Are you seeing this failure on ARM32?

Reply:
FWIW, `java/lang/instrument/GetObjectSizeIntrinsicsTest.java` passes on RPi 4 running ARM32 Linux.

Reply:
It's my fault, sorry, I forget to add you need to enable TieredCompilation explicitly, e.g.
`jtreg6.1/bin/jtreg -v1 -javaoptions:-XX:+TieredCompilation test/jdk/java/lang/instrument/GetObjectSizeIntrinsicsTest.java`

Reply:
Verified with the test. Hovewer the problem with logic_and should be investigated separately, since arm c1 assembler supports AND on 64bits values.